vm/src/matrix_3.h

```diff
@@ -0,0 +1,51 @@
+#ifndef BORGES_MATRIX_3_H
+#define BORGES_MATRIX_3_H
+
+#include "type.h"
+#include <initializer_list>
+
+namespace Borges {
+
+class Serializer;
+class Deserializer;
+
+class Matrix_3 {
+public:
+    class Const_row {
```

Comment:
as we are not implementing matrix math, the proxy class for the row seems overkill.